pkg/cloud/aws/services/kubeadm/configs.go

```diff
@@ -0,0 +1,152 @@
+
+// SetClusterConfigurationOptions will run all options on the passed in ClusterConfiguration.
+// Use this to customize an existing ClusterConfiguration.
+func SetClusterConfigurationOptions(base *kubeadmv1beta1.ClusterConfiguration, opts ...ClusterConfigurationOption) {
+	for _, opt := range opts {
+		opt(base)
+	}
+}
+
+// WithClusterName sets the cluster name on a ClusterConfiguration.
+func WithClusterName(name string) ClusterConfigurationOption {
+	return func(c *kubeadmv1beta1.ClusterConfiguration) {
+		c.ClusterName = name
+	}
+}
+
+// WithKubernetesVersion sets the kubernetes version on a ClusterConfiguration.
+func WithKubernetesVersion(version string) ClusterConfigurationOption {
+	return func(c *kubeadmv1beta1.ClusterConfiguration) {
+		c.KubernetesVersion = version
+	}
+}
+
+// WithAPIServerExtraArgs sets the kube-apiserver's extra arguments in the ClusterConfiguration.
+func WithAPIServerExtraArgs(args map[string]string) ClusterConfigurationOption {
+	return func(c *kubeadmv1beta1.ClusterConfiguration) {
+		if c.APIServer.ControlPlaneComponent.ExtraArgs == nil {
+			c.APIServer.ControlPlaneComponent.ExtraArgs = map[string]string{}
+		}
+		for key, value := range args {
+			c.APIServer.ControlPlaneComponent.ExtraArgs[key] = value
+		}
+	}
+}
+
+// WithControllerManagerExtraArgs sets the controller manager's extra arguments in the ClusterConfiguration.
+func WithControllerManagerExtraArgs(args map[string]string) ClusterConfigurationOption {
+	return func(c *kubeadmv1beta1.ClusterConfiguration) {
+		if c.ControllerManager.ExtraArgs == nil {
+			c.ControllerManager.ExtraArgs = map[string]string{}
+		}
+		for key, value := range args {
+			c.ControllerManager.ExtraArgs[key] = value
+		}
+	}
+}
+
+// WithClusterNetworkFromClusterNetworkingConfig maps a cluster api ClusterNetworkingConfig to a kubeadm Networking object.
+func WithClusterNetworkFromClusterNetworkingConfig(networkingConfig v1alpha1.ClusterNetworkingConfig) ClusterConfigurationOption {
+	return func(c *kubeadmv1beta1.ClusterConfiguration) {
+		c.Networking.DNSDomain = networkingConfig.ServiceDomain
+		c.Networking.PodSubnet = networkingConfig.Pods.CIDRBlocks[0]
+		c.Networking.ServiceSubnet = networkingConfig.Services.CIDRBlocks[0]
+	}
+}
+
+// WithControlPlaneEndpoint sets the control plane endpoint, usually the load balancer in front of the kube-apiserver(s).
+func WithControlPlaneEndpoint(endpoint string) ClusterConfigurationOption {
+	return func(c *kubeadmv1beta1.ClusterConfiguration) {
+		if c.ControlPlaneEndpoint == "" {
+			c.ControlPlaneEndpoint = endpoint
+		}
+	}
+}
+
+// WithAPIServerCertificateSANs will add the names provided to the kube-apiserver's list of certificate SANs.
+func WithAPIServerCertificateSANs(names ...string) ClusterConfigurationOption {
+	return func(c *kubeadmv1beta1.ClusterConfiguration) {
+		for _, name := range names {
+			c.APIServer.CertSANs = append(c.APIServer.CertSANs, name)
```

Comment:
I think `append(c.APIServer.CertSANs, names...)` could save some lines here? Not critical 🤷‍♂

Reply:
sure can! nice find